frame/nft/src/test/impls/mutate.rs

```diff
@@ -0,0 +1,235 @@
+/// Tests the implementation of [`Mutate::mint_into`].
+mod mint_into {
+	use std::collections::{BTreeMap, BTreeSet};
+
+	use composable_tests_helpers::test::helper::assert_last_event;
+	use composable_traits::financial_nft::NftClass;
+	use frame_support::{assert_noop, traits::tokens::nonfungibles::Mutate};
+	use sp_runtime::DispatchError;
+
+	use crate::{
+		pallet::*,
+		test::{
+			mock::{new_test_ext, Event, MockRuntime},
+			ALICE,
+		},
+	};
+
+	/// Tests minting an NFT into an account.
+	#[test]
+	fn success() {
+		new_test_ext().execute_with(|| {
+			const NEW_NFT_ID: NftInstanceId = 1;
+
+			Pallet::<MockRuntime>::mint_into(&NftClass::STAKING, &NEW_NFT_ID, &ALICE).unwrap();
+
+			assert_last_event::<MockRuntime>(Event::Nft(crate::Event::NftCreated {
+				class_id: NftClass::STAKING,
+				instance_id: NEW_NFT_ID,
+			}));
+
+			assert_eq!(
+				ClassInstances::<MockRuntime>::get(&NftClass::STAKING).unwrap(),
+				BTreeSet::from([NEW_NFT_ID]),
+				"STAKING class should only have one instance"
+			);
+
+			assert_eq!(
+				OwnerInstances::<MockRuntime>::get(&ALICE).unwrap(),
+				BTreeSet::from([(NftClass::STAKING, NEW_NFT_ID)]),
+				"ALICE should only have one instance"
+			);
+
+			assert_eq!(
+				Instance::<MockRuntime>::get(&(NftClass::STAKING, NEW_NFT_ID)).unwrap(),
+				(ALICE, BTreeMap::new()),
+				"owner should be ALICE, with no attributes"
+			);
+		})
+	}
+
+	/// Asserts that minting an NFT that already exists is an error.
+	#[test]
+	fn already_exists() {
+		new_test_ext().execute_with(|| {
+			const NEW_NFT_ID: NftInstanceId = 1;
+
+			Pallet::<MockRuntime>::mint_into(&NftClass::STAKING, &NEW_NFT_ID, &ALICE).unwrap();
+
+			assert_last_event::<MockRuntime>(Event::Nft(crate::Event::NftCreated {
+				class_id: NftClass::STAKING,
+				instance_id: NEW_NFT_ID,
+			}));
+
+			assert_noop!(
+				Pallet::<MockRuntime>::mint_into(&NftClass::STAKING, &NEW_NFT_ID, &ALICE),
+				DispatchError::from(crate::Error::<MockRuntime>::InstanceAlreadyExists)
+			);
+		})
+	}
+}
+
+/// Tests the implementation of [`Mutate::burn_from`].
+mod burn_from {
```

Comment:
afaik polkadot renamed its burn_from to burn, can rename later either as separate task

Reply:
Sounds like a problem for whoever does the upgrade 😁